Modules/Microsoft365DSC/DSCResources/MSFT_IntuneWindowsInformationProtectionPolicyWindows10MdmEnrolled/MSFT_IntuneWindowsInformationProtectionPolicyWindows10MdmEnrolled.psm1

```diff
@@ -166,11 +176,19 @@ function Get-TargetResource
                 $getValue = Get-MgBetaDeviceAppManagementMdmWindowsInformationProtectionPolicy `
                     -Filter "DisplayName eq '$DisplayName'" `
                     -ErrorAction SilentlyContinue
+                if ($getValue.count -gt 1)
+                {
+                    throw ("Error: Ensure the displayName {$displayName} is unique.")
+                }
+                if (-not [String]::IsNullOrEmpty($getValue.Id))
+                {
+                    $getValue = Get-MgBetaDeviceAppManagementMdmWindowsInformationProtectionPolicy -MdmWindowsInformationProtectionPolicyId $getValue.id -ExpandProperty assignments
```

Comment:
In worst case scenario this means it might take 3 separate calls to Graph to get our policy, maybe just expand the assignments property in previous call, remove ErrorAction or change it to Stop and add a try/catch there and this chunk won't be necessary?

Reply:
it has to be because you can't retrieve the navigation properties (assignments) when using filters